pkg/pii/anonymizer.go

```diff
@@ -13,24 +13,33 @@ type StringAnonymizer[K any] interface {
 	DeanonymizeString(key K, value string) (string, error)
 }
 
-type StructAnonymizer[K any] struct {
+type StructAnonymizer[T any, K any] struct {
 	stringAnonymizer StringAnonymizer[K]
 }
 
-func NewStructAnonymizer[K any](
+func NewStructAnonymizer[T any, K any](
```

Comment:
Do we expect `key` to be something different that a string?

Reply:
Not for event sourcing, but we plan to make `pii` a separate library, so it should be more abstract.